### PR TITLE
t2 run/push; implements tesselignore, eliminate t1 dep

### PR DESCRIPTION
**Pull request by rwaldron**
- updates existing tests
- removes now irrelevant test
- add new deps (fstream-ignore, tar)
- eliminates test/unit/common/tessel-classic.js
- makes `push` an option property, instead of its own parameter

### CODE REVIEW

File: lib/tessel/deploy.js [hunk 1]
```diff
@@ -52,89 +55,98 @@ Tessel.prototype.deployScript = function(opts, push) {
...
+          error += data.toString();
         });
 
-        // Tar up the code to improve transfer rates
-        tesselClassic.tarCode(ret.pushdir, {
-          node: true
-        }, function(err, bundle) {
-          if (err) {
-            return reject(err);
+        actions.tarBundle(process.cwd()).then(function(bundle) {
```
Comment (johnnyman727) [inline lib/tessel/deploy.js]: Should this be `process.cwd() + '/' + opts.entryPoint` as it was previously?

File: lib/tessel/deploy.js [hunk 2]
```diff
@@ -52,89 +55,98 @@ Tessel.prototype.deployScript = function(opts, push) {
...
+  });
+};
+
+actions.runScript = function(t, filepath, entryPoint) {
   logs.info('Running %s...', entryPoint);
   return new Promise(function(resolve) {
-    return t.connection.exec(commands.runScript(filepath, ret.relpath))
-      .then(function(remoteProcess) {
+    return t.connection.exec(commands.runScript(filepath, entryPoint))
+      .then(function(remote) {
```
Comment (johnnyman727) [inline lib/tessel/deploy.js]: Was `remoteProcess` too long for you? :stuck_out_tongue:

File: package.json [hunk 3]
```diff
@@ -59,7 +60,7 @@
     "sprintf-js": "^1.0.2",
     "ssh-keygen": "^0.2.1",
     "ssh2": "^0.4.2",
-    "tessel": "git+https://github.com/tessel/cli#jon-export-v2",
```
Comment (johnnyman727) [inline package.json]: Thank god.

File: lib/tessel/deploy.js (hunk 1 above)
Comment (johnnyman727) [inline lib/tessel/deploy.js]: I've confirmed that I can't deploy code from another directory than the one I'm in.

File: lib/tessel/deploy.js (hunk 2 above)
Comment (rwaldron) [inline lib/tessel/deploy.js]: Oh, I can change those back actually.

File: lib/tessel/deploy.js (hunk 1 above)
Comment (rwaldron) [inline lib/tessel/deploy.js]: > I've confirmed that I can't deploy code from another directory than the one I'm in.

Oh, I completely missed this use case.

**Comment by rwaldron** [comment]
@johnnyman727 ok, here's the output of a few different `t2 run ...` commands: 


- Run **file** from cwd: 
```
rwaldron at nova in ~/tbundle
$ t2 run default.js
INFO Connecting to Tessel...
INFO Connected to arnold over LAN
INFO Writing default.js to RAM on arnold (3.072 kB)...
INFO Deployed.
INFO Running default.js...
I'm blinking! (Press CTRL + C to stop)
I'm blinking! (Press CTRL + C to stop)
^CINFO Stopping script...
```

- Run **file** from some other directory: 
```
rwaldron at nova in ~/other-dir
$ t2 run ~/tbundle/default.js
INFO Connecting to Tessel...
INFO Connected to arnold over LAN
INFO Writing default.js to RAM on arnold (3.072 kB)...
INFO Deployed.
INFO Running default.js...
I'm blinking! (Press CTRL + C to stop)
I'm blinking! (Press CTRL + C to stop)
^CINFO Stopping script...
```


- Run **directory** from some other directory: 
```
rwaldron at nova in ~/other-dir
$ t2 run ~/tbundle/
INFO Connecting to Tessel...
INFO Connected to arnold over LAN
INFO Writing index.js to RAM on arnold (3.072 kB)...
INFO Deployed.
INFO Running index.js...
module.js:334
    throw err;
          ^
Error: Cannot find module '/tmp/remote-script/index.js'
    at Function.Module._resolveFilename (module.js:332:15)
    at Function.Module._load (module.js:282:25)
    at Function.Module.runMain (module.js:467:10)
    at startup (node.js:117:18)
    at node.js:948:3
INFO Stopping script...
```

This is _correct_ because there is no `index.js` file in that directory. 


- Run **directory** from other directory, assuming default name `index.js`: 
```
rwaldron at nova in ~/other-dir
$ cd ../tbundle/
rwaldron at nova in ~/tbundle
$ mv default.js index.js
rwaldron at nova in ~/tbundle
$ cd ../other-dir/
rwaldron at nova in ~/other-dir
$ t2 run ~/tbundle/
INFO Connecting to Tessel...
INFO Connected to arnold over LAN
INFO Writing index.js to RAM on arnold (3.072 kB)...
INFO Deployed.
INFO Running index.js...
I'm blinking! (Press CTRL + C to stop)
I'm blinking! (Press CTRL + C to stop)
^CINFO Stopping script...
```

- Run **directory** from other directory, assume using `"main": "app.js"` in `package.json`: 
```
rwaldron at nova in ~/other-dir
$ cd ../tbundle/
rwaldron at nova in ~/tbundle
$ mv index.js app.js
rwaldron at nova in ~/tbundle
$ cd ../other-dir/
rwaldron at nova in ~/other-dir
$ t2 run ~/tbundle/
INFO Connecting to Tessel...
INFO Connected to arnold over LAN
INFO Writing app.js to RAM on arnold (3.072 kB)...
INFO Deployed.
INFO Running app.js...
I'm blinking! (Press CTRL + C to stop)
I'm blinking! (Press CTRL + C to stop)
^CINFO Stopping script...
```

- Run **other directory** from other directory, which is actually empty: 

```
rwaldron at nova in ~/other-dir
$ t2 run ./
INFO Connecting to Tessel...
INFO Connected to arnold over LAN
WARN Invalid project directory
rwaldron at nova in ~/other-dir

$ t2 run ~/other-dir/
INFO Connecting to Tessel...
INFO Connected to arnold over LAN
WARN Invalid project directory

rwaldron at nova in ~/other-dir
$ t2 run /Users/rwaldron/other-dir/
INFO Connecting to Tessel...
INFO Connected to arnold over LAN
WARN Invalid project directory
```

- Run **non-existent file** from other directory, which is actually empty: 

```
rwaldron at nova in ~/other-dir
$ t2 run index.js
INFO Connecting to Tessel...
INFO Connected to arnold over LAN
WARN ENOENT, no such file or directory 'index.js'
```

**Comment by johnnyman727** [comment]
This looks fantastic! Very thorough. :clap: :clap:

**Pull request merged**